docs/book/in-depth-guide/laminas-db-sql-laminas-hydrator.md

```diff
@@ -398,7 +398,7 @@ use Laminas\Db\ResultSet\ResultSet;
 Next, we'll add the following import statements to our class file:
 
 ```php
-use Laminas\Hydrator\Reflection as ReflectionHydrator;
+use Laminas\Hydrator\ReflectionHydrator;
```

Comment:
@samsonasik 
It is the same change as provided in https://github.com/zendframework/tutorials/pull/112
I haven't merged this there, as it was still valid for hydrator v2.

We have the same issue here, and it is not noted that it is just for latest version of hydrator library.

Reply:
The tutorial takes the user through the steps of installing the latest version of Laminas. As a result the user gets deprecation notices when following the tutorial from the start. In my opinion that makes the framework look bad.

Reply:
I agreee with alex, most users uses latest components, as it is already laminas

Reply:
In general I do agree with @alexivaylov as well, that deprecation is bad when you follow the tutorial. But what is even worst is when it does not work.
In other components documentation is versioned, so it's not a problem. I am not sure what is the right approach in tutorials.

@froschdesign - what's your opinion? Thanks!

Reply:
@alexivaylov, @samsonasik 
> The tutorial takes the user through the steps of installing the latest version of Laminas.

Based on the existing PHP version and the skeleton application [works with version 7.1](https://github.com/laminas/laminas-mvc-skeleton/blob/26b6e06b107fa40f9678593203023e0a5a693618/composer.json#L13) of PHP, that means it can not use the latest version of laminas-hydrator. If someone uses PHP 7.1. then version 2.4.2 of laminas-hydrator will be installed.

An there is another problem:

> This tutorial assumes that you are running at least **PHP 5.6** with the Apache web server and MySQL, …

https://docs.laminas.dev/tutorials/getting-started/overview/#some-assumptions

---

A quick solution must now be found!

Reply:
PHP 7.1 and earlier versions are now EOL:
https://www.php.net/supported-versions.php

Reply:
@froschdesign Ok, so probably it require some paragraph redefinition to encourage user to uses latest php versions, otherwise, still uses `Laminas\Hydrator\Reflection`

Reply:
How about bumping the skeleton composer PHP version to "^7.2" and changing the documentation to reflect that?

Reply:
@alexivaylov
> PHP 7.1 and earlier versions are now EOL:
> https://www.php.net/supported-versions.php

I know this but it does not matter because the skeleton application can still be installed with version 7.1 of PHP.

@samsonasik 
> Ok, so probably it require some paragraph redefinition to encourage user to uses latest php versions, otherwise, still uses Laminas\Hydrator\Reflection

No, because the tutorial is already too complicated and this would not help a beginner.

@alexivaylov
> How about bumping the skeleton composer PHP version to "^7.2" and changing the documentation to reflect that?

This is the right direction!
It would be great if someone can create a pull request on [laminas-mvc-skeleton](https://github.com/laminas/laminas-mvc-skeleton).
Thanks in advance!

Reply:
@froschdesign I created PR https://github.com/laminas/laminas-mvc-skeleton/pull/17 for it.